src/node_code_integrity.cc

```diff
@@ -0,0 +1,237 @@
+    return env->ThrowError("Unable to open file");
+  }
+
+  const GUID wldp_host_other = WLDP_HOST_OTHER;
+  WLDP_EXECUTION_POLICY result;
+  HRESULT hr =
+      per_process::WldpCanExecuteFile(wldp_host_other,
+                                      WLDP_EXECUTION_EVALUATION_OPTION_NONE,
+                                      hFile,
+                                      NODEJS,
+                                      &result);
+  CloseHandle(hFile);
+
+  if (FAILED(hr)) {
+    return env->ThrowError("WldpCanExecuteFile failed");
+  }
+
+  bool isFileTrusted = (result == WLDP_EXECUTION_POLICY_ALLOWED);
+  args.GetReturnValue().Set(isFileTrusted);
+}
+
+static void IsInteractiveModeDisabledInternal(
+    const FunctionCallbackInfo<Value>& args) {
+  CHECK_EQ(args.Length(), 0);
+
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!per_process::isWldpInitialized) {
+    InitWldp(env);
+  }
+
+  if (per_process::WldpGetApplicationSettingBoolean != nullptr) {
+    BOOL ret;
+    HRESULT hr = per_process::WldpGetApplicationSettingBoolean(
+        NODEJS, DISABLE_INTERPRETIVE_MODE_SETTING_NAME, &ret);
+
+    if (SUCCEEDED(hr)) {
+      args.GetReturnValue().Set(Boolean::New(env->isolate(), ret));
+      return;
+    } else if (hr != E_NOTFOUND) {
+      // If the setting is not found, continue through to attempt
+      // WldpQuerySecurityPolicy, as the setting may be defined
+      // in the old settings format
+      args.GetReturnValue().Set(Boolean::New(env->isolate(), false));
+      return;
+    }
+  }
+
+  // WldpGetApplicationSettingBoolean is the preferred way for applications to
+  // query security policy values. However, this method only exists on Windows
+  // versions going back to circa Win10 2023H2. In order to support systems
+  // older than that (down to Win10RS2), we can use the deprecated
+  // WldpQuerySecurityPolicy
+  if (per_process::WldpQuerySecurityPolicy != nullptr) {
+    DECLARE_CONST_UNICODE_STRING(providerName, L"Node.js");
+    DECLARE_CONST_UNICODE_STRING(keyName, L"Settings");
+    DECLARE_CONST_UNICODE_STRING(valueName, L"DisableInteractiveMode");
+    WLDP_SECURE_SETTING_VALUE_TYPE valueType =
+        WLDP_SECURE_SETTING_VALUE_TYPE_BOOLEAN;
+    ULONG valueSize = sizeof(int);
+    int ret = 0;
+    HRESULT hr = per_process::WldpQuerySecurityPolicy(
+        &providerName, &keyName, &valueName, &valueType, &ret, &valueSize);
+    if (FAILED(hr)) {
+      args.GetReturnValue().Set(Boolean::New(env->isolate(), false));
+      return;
+    }
+
+    args.GetReturnValue().Set(
+        Boolean::New(env->isolate(), static_cast<bool>(ret)));
+  }
+}
+
+static void IsSystemEnforcingCodeIntegrity(
+    const FunctionCallbackInfo<Value>& args) {
+  CHECK_EQ(args.Length(), 0);
+
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!per_process::isWldpInitialized) {
+    InitWldp(env);
+  }
+
+  if (per_process::WldpGetApplicationSettingBoolean != nullptr) {
+    BOOL ret;
```

Comment:
Can you pick readable/maintainable variable names?

```suggestion
    bool ret;
```